src/test/FuzzerImpl.cpp

```diff
@@ -158,6 +184,93 @@ makeAssetCode(int digit)
     return code;
 }
 
+void
+generateStoredLedgerKeys(StoredLedgerKeys& keys)
+{
+    size_t constexpr NUM_UNVALIDATED_LEDGER_KEYS = 0x08U;
+    static_assert(NUM_UNVALIDATED_LEDGER_KEYS < NUM_STORED_LEDGER_KEYS,
+                  "No room for valid ledger keys");
+
+    auto const firstUnvalidatedLedgerKey =
+        NUM_STORED_LEDGER_KEYS - NUM_UNVALIDATED_LEDGER_KEYS;
+
+    for (auto index = 0; index < firstUnvalidatedLedgerKey; ++index)
+    {
+        keys[index] =
+            LedgerEntryKey(LedgerTestUtils::generateValidLedgerEntry());
+    }
+
+    for (auto index = firstUnvalidatedLedgerKey; index < NUM_STORED_LEDGER_KEYS;
+         ++index)
+    {
+        size_t const entrySize = 3;
+        keys[index] = autocheck::generator<LedgerKey>()(entrySize);
+    }
+}
+
+void
+storeLedgerKey(StoredLedgerKeys& keys, LedgerKey const& key)
+{
+    auto byte = getShortKey(key);
+    keys[byte % NUM_STORED_LEDGER_KEYS] = key;
+}
+
+void
+storeLedgerKeyIDsForOp(StoredLedgerKeys& keys,
+                       std::shared_ptr<OperationFrame const> opFrame)
+{
+    auto op = opFrame->getOperation();
+    auto const opType = op.body.type();
+
+    // For any operation type that we choose to intercept here, we can add to
+    // the set of stored LedgerKeys any LedgerKey that the operation makes
+    // available for reference by future operations, and give it a short ID by
+    // which future fuzz operations can refer to it (by calling getShortKey()).
+    switch (opType)
+    {
+    case CREATE_ACCOUNT:
+        storeLedgerKey(
+            keys,
+            accountKey(
+                opFrame->getOperation().body.createAccountOp().destination));
+        break;
+    case CREATE_CLAIMABLE_BALANCE:
+        storeLedgerKey(keys,
+                       claimableBalanceKey(opFrame->getResult()
+                                               .tr()
+                                               .createClaimableBalanceResult()
+                                               .balanceID()));
+        break;
+    case MANAGE_BUY_OFFER:
```

Comment:
> This probably doesn't do what you want it to do. To create an offer, offerID must be 0. You need to look at the result.

Perhaps a more compact way of doing this would be to change the code in `applySetupOperations()` so that it throws not only if a manage-offer operation returns `MANAGE_OFFER_DELETED`, but if it returns anything other than `MANAGE_OFFER_CREATED`.  It's only in `applySetupOperations()` that we call `storeLedgerKeyIDsForOp()`, so I think that would ensure that we were only storing offer keys for offers that had actually been created.  And I think it would make more sense with respect to setup semantics as well:  when we're setting up, we're just trying to establish some initial pre-fuzzing order book state, so I don't think that we mean to updating existing orders any more than we mean to be deleting them.

> Also, what about create passive offer?

Setup isn't creating any passive offers yet -- that's in PR #2927 .  Whichever of these PRs is merged first, the other will have to gain the chunk of code here to store keys for passive offers (and PR #2927 adds passive offers to the check for `MANAGE_OFFER_DELETED`, so if we also make the change in my comment above, then we'll have to merge those chunks too).

Reply:
Regarding passive offers, sure that's fine.

Regarding checking for `MANAGE_OFFER_CREATED`, I agree it would be better if you checked for `MANAGE_OFFER_CREATED`. But that alone still wouldn't fix this problem, right?

Reply:
Right, if I'm reading things correctly, I also need to use `opFrame->getResult().tr().manage(Buy|Sell)OfferResult().success().offer.offer().offerID)` in `storeLedgerKeyIDsForOp()`.